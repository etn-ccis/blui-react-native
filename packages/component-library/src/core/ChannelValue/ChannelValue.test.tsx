import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ChannelValue } from '.';

describe('ChannelValue', () => {
    it('ChannelValue Renders', async () => {
        let renderer: TestRenderer.ReactTestRenderer;
        await TestRenderer.act(() => {
            renderer = TestRenderer.create(<ChannelValue value="2" />);
        });
        expect(renderer!.toJSON()).toMatchSnapshot();
    });

    it('Renders the background color and font color correctly', async () => {
        let renderer: TestRenderer.ReactTestRenderer;
        await TestRenderer.act(() => {
            renderer = TestRenderer.create(
                <ChannelValue
                    value="50"
                    units="%"
                    icon={{ family: 'material-community', name: 'chart-pie' }}
                    iconColor="red"
                />
            );
        });
        expect(renderer!.toJSON()).toMatchSnapshot();
    });

    it('Accepts style override', async () => {
        let renderer: TestRenderer.ReactTestRenderer;
        await TestRenderer.act(() => {
            renderer = TestRenderer.create(<ChannelValue value="2" style={{ backgroundColor: 'red' }} />);
        });
        expect(renderer!.toJSON()).toMatchSnapshot();
    });
});
