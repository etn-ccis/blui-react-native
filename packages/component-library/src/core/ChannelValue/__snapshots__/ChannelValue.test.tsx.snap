// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ChannelValue Accepts style override 1`] = `[Function]`;

exports[`ChannelValue ChannelValue Renders 1`] = `null`;

exports[`ChannelValue Renders the background color and font color correctly 1`] = `[Function]`;
